File: Maximum Depth Of Binary Tree/MaxDepthOfBinaryTree.cs
using System;

int MaxDepth(TreeNode root) 
{
    return MaxDepth(root, 0);
}

int MaxDepth(TreeNode node, int level)
{
    if(node == null) 
    {
        return 0;
    }        
    
    level++;
    
    int max = level;        
    
    if(node.left != null)
    { 
        var temp = MaxDepth(node.left, level);
        max = Math.Max(temp, max);
    }
    
    if(node.right != null)
    {
        var temp = MaxDepth(node.right, level);
        max = Math.Max(temp, max);
    }
    
    return max;
}

File: Maximum Depth Of Binary Tree/MaxDepthOfBinaryTree.java
class Solution {
    public int maxDepth(TreeNode root) {
        return maxDepth(root, 0);
    }
    
    public int maxDepth(TreeNode root, int depth) {
        if(root == null) return 0;
        
        return Math.max(maxDepth(root.left, depth), maxDepth(root.right, depth)) + 1;
    }
}